streampipes-extensions/streampipes-processors-transformation-jvm/src/test/java/org/apache/streampipes/processors/transformation/jvm/Additional Tests

```diff
@@ -0,0 +1,96 @@
+import org.apache.streampipes.commons.exceptions.SpRuntimeException;
+import org.apache.streampipes.extensions.api.pe.context.EventProcessorRuntimeContext;
+import org.apache.streampipes.extensions.api.pe.routing.SpOutputCollector;
+import org.apache.streampipes.model.runtime.Event;
+import org.apache.streampipes.wrapper.params.compat.ProcessorParams;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+public class TextFilterProcessorTest {
+
+    @Mock
+    private ProcessorParams processorParams;
+
+    @Mock
+    private SpOutputCollector spOutputCollector;
+
+    @Mock
+    private EventProcessorRuntimeContext eventProcessorRuntimeContext;
+
+    @BeforeEach
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void testConstructorInitialization() {
+        TextFilterProcessor processor = new TextFilterProcessor();
+        assertNull(processor.getKeyword());
+        assertNull(processor.getStringOperator());
+        assertNull(processor.getFilterProperty());
+    }
+
+    @Test
+    public void testModelDeclaration() {
+        TextFilterProcessor processor = new TextFilterProcessor();
+        assertNotNull(processor.declareModel());
+        // Add more assertions to validate the description
+    }
+
+    @Test
+    public void testOnInvocation() {
+        TextFilterProcessor processor = new TextFilterProcessor();
+        when(processorParams.extractor().singleValueParameter(anyString(), eq(String.class))).thenReturn("keyword");
+        when(processorParams.extractor().selectedSingleValue(anyString(), eq(String.class))).thenReturn("MATCHES");
+        when(processorParams.extractor().mappingPropertyValue(anyString())).thenReturn("text");
+
+        assertDoesNotThrow(() -> processor.onInvocation(processorParams, spOutputCollector, eventProcessorRuntimeContext));
+
+        assertEquals("keyword", processor.getKeyword());
+        assertEquals(TextFilterProcessor.StringOperator.MATCHES, processor.getStringOperator());
+        assertEquals("text", processor.getFilterProperty());
+    }
+
+    @Test
+    public void testOnEvent() {
+        TextFilterProcessor processor = new TextFilterProcessor();
+        when(processorParams.extractor().singleValueParameter(anyString(), eq(String.class))).thenReturn("keyword");
+        when(processorParams.extractor().selectedSingleValue(anyString(), eq(String.class))).thenReturn("MATCHES");
+        when(processorParams.extractor().mappingPropertyValue(anyString())).thenReturn("text");
+
+        processor.onInvocation(processorParams, spOutputCollector, eventProcessorRuntimeContext);
+
+        Event matchingEvent = createMockEvent("Some text that MATCHES the keyword");
```

Comment:
This text only contains the keyword, doesn't it?
If you would like to test for equality, your string should just be `keyword`.